File: easymall/src/main/java/org/ruikun/entity/Product.java
package org.ruikun.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("product")
public class Product {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;

    private String subtitle;

    private String description;

    private BigDecimal originalPrice;

    private BigDecimal price;

    private Integer stock;

    private Integer sales;

    private String image;

    private String images; // 多个图片，逗号分隔

    private Long categoryId;

    private String brand;

    private Integer status; // 0-下架 1-上架

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    @TableField(fill = FieldFill.INSERT)
    private Integer deleted;

    @TableField(exist = false)
    private String categoryName;
}

File: easymall-frontend/legacy-vue/src/types/product.ts
export interface ProductPageItem {
  id: number
  name: string
  subtitle: string
  price: number
  stock: number
  sales: number
  image: string
  categoryId: number
  categoryName: string
  status: number
  createTime: string
}

export interface ProductDetail {
  id: number
  name: string
  subtitle: string
  description: string
  originalPrice: number
  price: number
  stock: number
  sales: number
  image: string
  images: string[]
  categoryId: number
  categoryName: string
  brand: string
  status: number
  createTime: string
  updateTime: string
}

export interface ProductFormData {
  name: string
  subtitle: string
  description: string
  originalPrice: number | null
  price: number | null
  stock: number | null
  image: string
  images: string
  categoryId: number | null
  brand: string
}

export interface ProductQuery {
  pageNum: number
  pageSize: number
  name?: string
  categoryId?: number
  status?: number
}
